File: source/code/csharp/authentication/create_directory_fb.cs
using Stormpath.SDK.Provider;

var facebookDirectory = client.Instantiate<IDirectory>();
facebookDirectory.SetName("My Facebook Directory");

await client.CreateDirectoryAsync(facebookDirectory,
    options => options.ForProvider(
        client.Providers()
            .Facebook()
            .Builder()
            .SetClientId("YOUR_FACEBOOK_APP_ID")
            .SetClientSecret("YOUR_FACEBOOK_APP_SECRET")
            .Build()
        ));

File: source/code/java/authentication/create_directory_fb.java
Directory directory = client.instantiate(Directory.class);
directory.setName("My Facebook Directory");
directory.setDescription("A Facebook directory");

CreateDirectoryRequest request = Directories.newCreateRequestFor(directory)
    .forProvider(Providers.FACEBOOK.builder()
        .setClientId("YOUR_FACEBOOK_APP_ID")
        .setClientSecret("YOUR_FACEBOOK_APP_SECRET")
        .build()
    ).build();

directory = client.createDirectory(request);
